play-json/shared/src/main/scala/Json.scala

```diff
@@ -414,6 +419,29 @@ object Json extends JsonFacade {
   sealed trait MacroOptions
 
   /**
+   * Defines the default macro options.
+   *
+   * Since you can't have defaults for a type parameter (unless it's a contravariant type, it will default to
+   * Nothing), we supply the default via an implicit parameter.
+   */
+  trait DefaultMacroOptions[O <: Json.MacroOptions]
+
+  /**
+   * Low priority implicit used when Json.MacroOptions is not passed.
```

Comment:
I think the naming is bothering me. If I understand right, it's more an evidence typeclass, to avoid ambiguity around the `MacroOptions` type (could be manage with 2.12 `@implicitNotFound`?), rather that a typeclass defining by itself such options?